### ISSUE TITLE
Behaviour for direct consumer with ConsumePartitions() seems to have changed in v1.11.5?

### ISSUE THREAD

**Issue by edocsss** [twmb/franz-go#337]
Hi, I am currently using `franz-go` for a direct Kafka consumer with a specific partition and offset.

Below is the way I initialise the client:

```go
client, err := kgo.NewClient(
	kgo.SeedBrokers(brokers...),
	kgo.ConsumePartitions(map[string]map[int32]kgo.Offset{
		"topic_name": {int32(0): 1000},
	}),
)
```

In v1.11.3, this works perfectly fine. If I'm asking for the consumer to read from `topic_name` partition `0` starting from offset `1000`, it will do exactly that.

However, in v1.11.5, this behaviour seems to have changed. The same code above will cause the consumer to read from the following partitions (assume 5 partitions in total):

- Partition 0 -> starting from offset 1000.
- Partition 1 -> starting from offset -2.
- Partition 2 -> starting from offset -2.
- Partition 3 -> starting from offset -2.
- Partition 4 -> starting from offset -2.

This means this direct consumer now reads from all partitions, instead of just the assigned partitions and offsets in `kgo.ConsumePartitions()`.

Based on my own reading, looks like [this code changes](https://github.com/twmb/franz-go/compare/v1.11.3...v1.11.5#diff-65862fa6dc5957b7e17dec29a07ad15e4b36b64f8748e181d50421064b71663dL64) made the issue.

In v1.11.3, `useTopic` is always `false` if I only use `ConsumePartitions`, but not `ConsumeTopics`, and as a result, only the topics, partitions, and offsets supplied in `ConsumePartitions` is being assigned to this consumer.

In v1.11.5, looks like `useTopic` is always `true` even though I'm not passing `ConsumeTopics`. The following code will be triggered in v1.11.5 (but not v1.11.3):

```go
if useTopic {
	partitions := topicPartitions.load()
	if d.cfg.regex && partitions.isInternal {
		continue
	}

        // This causes the `toUse` for the topic to contain all partitions
        // with -2 (or the d.cfg.resetOffset value).
	toUseTopic := make(map[int32]Offset, len(partitions.partitions))
	for partition := range partitions.partitions {
		toUseTopic[int32(partition)] = d.cfg.resetOffset
	}

	toUse[topic] = toUseTopic
}
```

I wonder if this change in behaviour is expected? If so, how can I correctly just consume from the specific partition from the given offset in v1.11.5?

Thanks!

**Comment by twmb** [comment]
I see, yes that was inadvertent. I can fix this Sun or Mon and this should
be easy enough to add an integration test for. I recommend staying on
1.11.3 until then, my mistake.

**Comment by edocsss** [comment]
Hi @twmb noted on the timeline, thanks for the support! And yes, will stick to 1.11.3 for now.

**Comment by twmb** [comment]
This has been patched and tested in v1.11.6
